src/Nox.Types/Types/Weight/Weight.cs

```diff
@@ -1,51 +1,141 @@
 using Nox.Types.Common;
+
+using System;
 using System.Collections.Generic;
+using System.Globalization;
 
 namespace Nox.Types;
 
 /// <summary>
 /// Represents a Nox <see cref="Weight"/> type and value object.
 /// </summary>
-public class Weight : Measurement<Weight, WeightUnit>
+public class Weight : ValueObject<QuantityValue, Weight>
 {
+    public const int QuantityValueDecimalPrecision = 6;
```

Comment:
Since the internals are visible to SqlServer project, can we mark this field as internal?